purchase_last_price_info/models/product_product.py

```diff
@@ -53,10 +53,15 @@ def _compute_last_purchase_line_id(self):
     @api.depends("last_purchase_line_id")
     def _compute_last_purchase_line_id_info(self):
         for item in self:
+            currency = (
+                item.last_purchase_line_id.currency_id
+                or item.company_id.currency_id
+                or item.currency_id
```

Comment:
```suggestion
                or item.currency_id
```

I think you should favour the product currency and stop there.
As the field is required it'll always be set